examples/mem-alloc-test/src/main.rs

```diff
@@ -0,0 +1,152 @@
+use core::arch::global_asm;
+use std::{thread};
+use std::time::{Instant, Duration};
+use rand::Rng;
+use std::sync::{Mutex, Arc};
+
+
+enum SizeType {
+    LARGE_MEM,
+    MEDIUM_MEM,
+    LOW_MEM,
+}
+
+fn calculateAndPrintStat(shared_mutex_clone: Arc<Mutex<i32>>,
+                         shared_mutex_dur_clone: Arc<Mutex<u128>>,
+                         duration: Duration, tid: i32, memtypestr: &str) {
+
+    /* This function continuosly prints the average execution time of the worker
+     * thread which includes time to allocate a buffer + time to traverse it +
+     * time to free it.
+     * TODO: Currently the average is taken over all threads (large, medium and
+     * small). I will a separate average timings for small, medium and large
+     * buffer threads
+     */
+
+    let duration_ms =   duration.as_nanos();
+    let mut data = shared_mutex_clone.lock().unwrap();
+    *data += 1;
+    let mut data_d = shared_mutex_dur_clone.lock().unwrap();
+    *data_d += duration_ms;
+    let avg_exec_time = *data_d as f64 / *data as f64;
+
+    if ((*data % 10001 == 0 && memtypestr == "LOW") ||
+        (*data % 1001 == 0 && memtypestr == "MEDIUM") ||
+        (*data % 12 == 0 && memtypestr == "LARGE")) {
+        println!("Thread {} ({})records count {} avg execution time {}", tid,
+                 memtypestr, *data, avg_exec_time);
+    }
+}
+
+
+fn Get_Random_Num(start: usize, end: usize) -> usize {
+    let mut rng = rand::thread_rng();
+    let random_number: usize = rng.gen_range(start..=end);
+    return random_number;
+}
+
+fn worker_thread(tid: i32, shared_mutex_clone: Arc<Mutex<i32>>,
+                 shared_mutex_dur_clone: Arc<Mutex<u128>>, memtype: SizeType) {
+    loop {
+        let mut page_size = 4096;
+        let to_kb = 1024;
+        let to_mb = to_kb * 1024;
+        let to_gb = to_mb * 1024;
+
+        let mut ran_num = 0;
+        let mut size = 0;
+        let mut scan_interval = page_size;
+        let memtypestr;
```

Comment:
The rust style for this would be to have each arm of the `match` produce the values you need (as a tuple, since this is producing multiple values). So something like:

```
let (scan_interval, size, memtypestr) = match memtype {
    SizeType::LARGE_MEM => {
         (Get_Random_Num(1, 4) * to_gb, String::from("LARGE"))
    },
    SizeType::MEDIUM_MEM => {
    }
   ...
};
```

Basically in Rust, things like `match` and `if` can evaluate to a value, so you can do things like `let x = if ...` or `let y = match ...`. This form is generally preferred over assigning a variable in each arm of the `match` or `if`. Rust also has native support for tuples.

Also, I think there's a better way of getting a string from the enumeration names than having explicit code here that does it. In this particular case, it looks like we only care about the strings when we're printing the output. In that case, we can just use the enumeration value everyewhere. You can add `#[derive(Debug)]` on the line before your `enum SizeType` definition. This tells the Rust compiler to automatically generate code for a debug representation of the type. In the case of enums, this is just the name of the Enum variant. When you print the results, if you print using `"{:?}"` instead of just `"{}", it will print using the debug representation. So something like ```fun myfun(var: SizeType) { println!("{:?}", var); }``` will just do what you want, without having to write explicit conversions to string for your enumeration.

Reply:
If you do need the string representation at run-time for other reasons, then the Rust-y way to do it would be to add a `to_str()` method to the enumeration, so the conversion is tied to the enumeration type. Rust doesn't have classes and objects, but you can add functions to enums and structs that operate somewhat like methods on classes in object-oriented language.